client/src/views/auth/Signin.vue

```diff
@@ -30,9 +30,11 @@ export default class Signin extends Mixins(NrAffiliationMixin) {
   /** called when keycloak session is ready. */
   private async onReady () {
     if (this.redirectUrl) {
+      await this.$router.push(this.redirectUrl)
+
       // If there is stored NR data to process, create the affiliation OR navigate to redirectUrl
       const nr = JSON.parse(sessionStorage.getItem('NR_DATA'))
-      nr ? await this.createAffiliation(nr) : await this.$router.push(this.redirectUrl)
+      if (nr) await this.createAffiliation(nr)
```

Comment:
@severinbeauvais I had to change this back. Signin wouldn't resolve back to root in the event that the business process failed. So it's safer to process the redirect first, and then handle the business processing afterwards :)